src/DotNetWorker.Core/Converters/CancellationTokenConverter.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Functions.Worker.Converters
+{
+    internal class CancellationTokenConverter : IInputConverter
+    {
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            if (context.TargetType == typeof(CancellationToken))
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Success(context.FunctionContext.CancellationToken));
```

Comment:
We cannot get the cancellation token from the invocation, so we need to flow it through to the function context as it's public

Reply:
May be we should update the if to support `CancellationToken?` type as well. Customer can add `CancellationToken? token` param to the function. See the GuidConverter.